module/VuFind/src/VuFind/View/Helper/Root/MakeLink.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Make link view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+/**
+ * Make link view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class MakeLink extends MakeTag
```

Comment:
I'm concerned that extending `MakeTag` may lead to problems down the road as we get more explicit about typing, because the `__invoke()` signatures are significantly different, and PHP is getting increasingly strict about signature compatibility between classes and subclasses. I would propose two possible options:

1.) Make an `AbstractTagBuilder` base class, extend that with `MakeLink` and `MakeTag`, and then define `__invoke` differently in each of those subclasses. This way there is no possible conflict in inheritance.

2.) Instead of using inheritance, make both helpers stand-alone; revise the `__invoke()` method of `MakeLink` so that it formats arguments to then call `$this->getView()->plugin('makeTag')()`.

Reply:
I decided to go with option 2.

Unless I'm mistaken, an abstract class would still require a signature on __invoke and I wasn't sure about the Abstract being a view helper if it was just a utility helper, etc.